### PR TITLE
Add LayerGoString() to packet.go

### PR DESCRIPTION
**Pull request by randstr**
And minor fixups.

### CODE REVIEW

File: layers/ip6.go [hunk 1]
```diff
@@ -399,8 +334,14 @@ func serializeTLVOptionPadding(data []byte, padLength int) {
 	return
 }
 
-func serializeIPv6HeaderTLVOptions(buf []byte, options []*ipv6HeaderTLVOption, fixLengths bool, dryrun bool) int {
+// If buf is 'nil' do a dryrun 
+func serializeIPv6HeaderTLVOptions(buf []byte, options []*ipv6HeaderTLVOption, fixLengths bool) int {
 	var l int
+	var dryrun bool
```
Comment (gconnell) [inline layers/ip6.go]: dryrun := buf == nil

**Comment by gconnell** [comment]
Thanks for the code!  Just a few minor issues, and we should be good to go.

**Comment by randstr** [comment]
Nice catch. Please confirm fix. And thanks for the review.

File: packet.go [hunk 2]
```diff
@@ -300,6 +301,91 @@ func layerString(i interface{}, anonymous bool, writeSpace bool) string {
...
+
+func layerGoString(i interface{}, b *bytes.Buffer) {
+	if s, ok := i.(fmt.GoStringer); ok {
+		b.WriteString(s.GoString())
+		return
+	}
+
+	var v reflect.Value
+	var ok bool
+	if v, ok = i.(reflect.Value); !ok {
```
Comment (gconnell) [inline packet.go]: general style nit, optional:

    v, ok := i.(reflect.Value)
    if !ok {
      ...
    }

**Pull request merged**